test/common/router/scoped_rds_test.cc

```diff
@@ -1453,15 +1453,17 @@ on_demand: true
                             Stats::Gauge::ImportMode::Accumulate)
                      .value());
   // All the on demand updated callbacks will be executed when the route table comes.
-  for (int i = 0; i < 5; i++) {
+  {
     ScopeKeyPtr scope_key = getScopedRdsProvider()->config<ScopedConfigImpl>()->computeScopeKey(
         TestRequestHeaderMapImpl{{"Addr", "x-foo-key;x-foo-key"}});
-    std::function<void(bool)> route_config_updated_cb = [](bool) {};
+    std::function<void(bool)> route_config_updated_cb = [](bool scope_exist) {
+      EXPECT_TRUE(scope_exist);
+    };
     getScopedRdsProvider()->onDemandRdsUpdate(std::move(scope_key), event_dispatcher_,
                                               std::move(route_config_updated_cb));
   }
   // After on demand request, push rds update, the callbacks will be executed.
-  EXPECT_CALL(event_dispatcher_, post(_)).Times(5);
+  EXPECT_CALL(event_dispatcher_, post(_)).Times(1);
```

Comment:
here you want to capture the arg by saving the arg, and call it.

xxxx.WillOnce(SaveArg<0>(&post_cb));
post_cb();

so that you can verify the expect at 1460 is evaluated.

Reply:
The callback will be called after it is posted, no need to say it and execute it manually

Reply:
ah, right mockdispatcher does do that.